sn_networking/src/lib.rs

```diff
@@ -621,6 +622,43 @@ impl Network {
             .await)
     }
 
+    pub async fn get_store_cost_from_network(&self, dbc_id_key: NetworkAddress) -> Result<Token> {
+        let (sender, receiver) = oneshot::channel();
+        debug!("Attempting to get store cost");
+        // first we need to get CLOSE_GROUP of the dbc_id
+        self.send_swarm_cmd(SwarmCmd::GetClosestPeers {
+            key: dbc_id_key.clone(),
+            sender,
+        })?;
+
+        let close_nodes = receiver
+            .await
+            .map_err(|_e| Error::InternalMsgChannelDropped)?
+            .into_iter()
+            .collect_vec();
+
+        let request = Request::Query(Query::GetStoreCost(dbc_id_key));
+        let responses = self
+            .send_and_get_responses(close_nodes, &request, true)
+            .await;
+
+        // loop over responses, generating an avergae fee and storing all responses along side
+        let mut all_costs = vec![];
+        for response in responses.into_iter().flatten() {
```

Comment:
in case there is error among the responses, say due to time_out, then the `flatten` will just swallow the error right?
in that case, shall we ensure the final OK responses still meet a minium len, say more than 3 ?

Reply:
That's handled in the `get_fee_from_storecost_quotes`